gdal/doc/source/drivers/raster/jp2grok.rst

```diff
@@ -0,0 +1,357 @@
+GEOREF_SOURCES open option) whose value is a comma-separated list of the
+following keywords : PAM, GEOJP2, GMLJP2, INTERNAL (shortcut for
+GEOJP2,GMLJP2), WORLDFILE, NONE. First mentioned sources are the most
+prioritary over the next ones. A non mentioned source will be ignored.
+
+For example setting it to "WORLDFILE,PAM,INTERNAL" will make a
+geotransformation matrix from a potential worldfile prioritary over PAM
+or internal JP2 boxes. Setting it to "PAM,WORLDFILE,GEOJP2" will use the
+mentioned sources and ignore GMLJP2 boxes.
+
+Thread support
+--------------
+
+By default, if the JPEG2000 file has internal tiling, GDAL will try to
+decode several tiles in multiple threads if the RasterIO() request it
+receives intersect several tiles. This behavior can be controlled with
+the GDAL_NUM_THREADS configuration option that defaults to ALL_CPUS in
+that context. In case RAM is limited, it can be needed to set this
+configuration option to 1 to disable multi-threading
+
+Multi-threaded decoding can also be
+enabled at the code-block level. This must be enabled with the
+OPJ_NUM_THREADS environment variable (note: this is a system environment
+variable, not a GDAL configuration option), which can be set to a
+numeric value or NUM_CPUS. Its default value is 1. Starting with GDAL
+2.3, this multi-threading at code-block level is automatically enabled
+by GDAL.
+Starting with GDAL 3.0.1, multi-threaded encoding is
+automatically enabled and can be controlled with the OPJ_NUM_THREADS environment
+variable or the GDAL_NUM_THREADS configuration option.
+
+Both multi-threading mechanism can be combined together.
+
+Option Options
+--------------
+
+The following open option is available:
+
+-  **1BIT_ALPHA_PROMOTION=YES/NO**: Whether a 1-bit alpha channel should
+   be promoted to 8-bit. Defaults to YES.
+
+-  **GEOREF_SOURCES=string**: (GDAL > 2.2) Define which georeferencing
+   sources are allowed and their priority order. See
+   `Georeferencing <#georeferencing>`__ paragraph.
+
+-  **USE_TILE_AS_BLOCK=YES/NO**: (GDAL > 2.2) Whether to always use the
+   JPEG-2000 block size as the GDAL block size Defaults to NO. Setting
+   this option can be useful when doing whole image decompression and
+   the image is single-tiled. Note however that the tile size must not
+   exceed 2 GB since that's the limit supported by GDAL.
+
+Creation Options
+----------------
+
+-  **CODEC=JP2/J2K** : JP2 will add JP2 boxes around the codestream
+   data. The value is determined automatically from the file extension.
+   If it is neither JP2 nor J2K, J2K codec is used.
+
+-  **GMLJP2=YES/NO**: Indicates whether a GML
+   box conforming to the OGC GML in JPEG2000 specification should be
+   included in the file. Unless GMLJP2V2_DEF is used, the version of the
+   GMLJP2 box will be version 1. Defaults to YES.
+-  **GMLJP2V2_DEF=filename**: Indicates whether
+   a GML box conforming to the `OGC GML in JPEG2000, version
```

Comment:
The "GMLJP2v2 Definition File" section found in jp2openjpeg.rst is omitted in this rst, and seems relevant to the use of this creation option.  Perhaps we should reference the jp2openjpeg.rst file's section?

Reply:
Thanks, fixed. As I had to remove a few files based on your other comments, I have squashed all of the changes back into a single commit. Be warned - you will need to do a `git fetch && git reset --hard origin/grok` to pull the latest.